geometry/perspective.hpp

```diff
@@ -32,9 +33,17 @@ class Perspective final {
   std::experimental::optional<RP2Point<Scalar, ToFrame>> operator()(
       Point<Vector<Scalar, FromFrame>> const& point) const;
 
+  // Returns true iff the |point| is hidden by the |sphere| in this perspective.
   bool IsHiddenBySphere(Point<Vector<Scalar, FromFrame>> const& point,
                         Sphere<Scalar, FromFrame> const& sphere) const;
 
+  // Returns the (sub)segments of |segment| that are visible in this perspective
+  // after taking into account the hiding by |sphere|.  The returned vector has
+  // 0, 1, or 2 elements.
+  std::vector<Segment<Point<Vector<Scalar, FromFrame>>>> VisibleSegments(
```

Comment:
Also, as discussed, this should *really* use Positions anyway, it's getting way too clumsy for no good reason (a job for another CL though).